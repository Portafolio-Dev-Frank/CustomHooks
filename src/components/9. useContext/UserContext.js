import { createContext } from 'react';

// HIGH ORDER COMPONENT 
export const UserContext = createContext(null);

