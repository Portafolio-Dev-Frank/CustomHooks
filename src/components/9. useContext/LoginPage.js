import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginPage = () => {

    const { user, setUser } = useContext(UserContext)

    const quemado = {
        id: 1234,
        nombre: 'juancho',
        email: 'juan@rmail.email.co'
    }

    return (
        <div>
            <h1>LoginPage</h1>
            <hr></hr>
            <button className="btn btn-primary"
                onClick={() => setUser(quemado)}>
                    Login
            </button>
        </div>
    )
}
