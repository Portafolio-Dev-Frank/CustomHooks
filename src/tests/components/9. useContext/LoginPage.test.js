import { shallow } from "enzyme"
import { LoginPage } from "../../../components/9. useContext/LoginPage"
import { UserContext } from "../../../components/9. useContext/UserContext"


describe('pruebas en <LoginPage></LoginPage>', () => {

    const user = {
        name: 'francisco',
        email: 'francisco.email.co'
    }
    const setUser = jest.fn();
    // para que las pruebas pasen debe ser mount y no shallow
    const wrapper = shallow(
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <LoginPage></LoginPage>
        </UserContext.Provider>)

    test('debe de renderizarse el componente LoginPage', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de ejecutar el setUser con el argumento esperado  id: 1234 nombre: juancho', () => {
        
        wrapper.find('button').simulate('click');
        expect(setUser).toHaveBeenCalledTimes(1);
    })
    
    
})
