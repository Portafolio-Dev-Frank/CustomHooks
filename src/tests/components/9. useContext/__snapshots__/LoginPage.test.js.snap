// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <LoginPage></LoginPage> debe de renderizarse el componente LoginPage 1`] = `<LoginPage />`;
